py3status/modules/updates.py

```diff
@@ -0,0 +1,215 @@
+    {yay}     number of updates, eg 0 .. Arch Linux (AUR)
+    {zypper}  number of updates, eg 0 .. openSUSE         .. NOT TESTED
+
+Color thresholds:
+    xxx: print a color based on the value of `xxx` placeholder
+
+@author Iain Tatch <iain.tatch@gmail.com> (arch_updates)
+@author Joshua Pratt <jp10010101010000@gmail.com> (apt_updates)
+@license BSD (arch_updates), BSD (apt_updates)
+
+Examples:
+```
+# multiple distributions, same format
+updates {
+    format = '[\?not_zero Updates [\?color=update {update}]]'
+}
+
+# Arch Linux
+updates {
+    format = 'Updates [\?color=pacman {pacman}]/[\?color=auracle {auracle}]'
+    # format = 'Updates [\?color=pacman {pacman}]/[\?color=yay {yay}]'
+}
+```
+
+SAMPLE OUTPUT
+[{'full_text': 'Apk '}, {'full_text': '3', 'color': '#a9a9a9'}]
+
+14pkgs
+[{'full_text': 'Apt '}, {'full_text': '14', 'color': '#00FF00'}]
+
+29and5pkgs
+[{'full_text': 'Pacman '}, {'full_text': '29 ', 'color': '#FFFF00'},
+{'full_text': 'Auracle '}, {'full_text': '5', 'color': '#a9a9a9'}]
+
+35pkgs
+[{'full_text': 'Zypper '}, {'full_text': '34', 'color': '#ffa500'}]
+
+45pkgs
+[{'full_text': 'Xbps '}, {'full_text': '45', 'color': '#FF0000'}]
+
+no_updates
+{'full_text': 'No Updates'}
+"""
+
+STRING_INVALID = 'invalid manager'
+
+
+class Update:
+    def __init__(self, parent, name, command):
+        self.parent = parent
+        self.name = name
+        self.command = command
+
+    def get_output(self):
+        try:
+            return self.parent.py3.command_output(self.command)
+        except self.parent.py3.CommandError:
+            return None
+
+    def count_updates(self, output):
+        return len(output.splitlines())
+
+    def get_updates(self):
+        output = self.get_output()
+        if output is None:
+            return {self.name: output}
+        return {self.name: self.count_updates(output)}
+
+
+class Apt(Update):
+    def count_updates(self, output):
+        return len(output.splitlines()[1:])
+
+
+class Apk(Update):
+    def count_updates(self, output):
+        return len(output.splitlines()[1:])
+
+
+class Eopkg(Update):
+    def get_output(self):
+        output = self.parent.py3.command_output(self.command)
+        if "No packages to upgrade." in output:
+            return ""
+        return output
+
+
+class Zypper(Update):
+    def count_updates(self, output):
+        return len([x for x in output.splitlines() if x][4:])
+
+
+class Py3status:
+    """
+    """
+
+    # available configuration parameters
+    cache_timeout = 600
+    format = None
+    thresholds = [
+        (0, 'darkgray'), (10, 'degraded'), (20, 'orange'), (30, 'bad')]
+
+    class Meta:
+        deprecated = {
+            "rename_placeholder": [
+                {
+                    "placeholder": "aur",
+                    "new": "auracle",
+                    "format_strings": ["format"],
+                }
+            ]
+        }
+
+    def post_config_hook(self):
+        managers = update_managers = [
```

Comment:
Can we avoid two vars? Not sure which one user is expected to modify.

Reply:
I need two vars. In if-else below, managers are being replaced with custom managers (if given). 
I have code there to compare custom managers against update_managers. In the end, they all become managers. `managers` here will be used if there are no custom managers.

Reply:
This was literally added to make what you want with `managers` + `format`.

Reply:
Not sure what you refer to, but the only usage of `update_managers` is in the `else` block of `if isinstance(custom, tuple):` and all I'm saying is that I'm not fond of that alternative way of configuring `managers` — in my mind it should only support configuration via list of tuples

Reply:
I disagree, debug configs are not defined by how many people use them, just like `thresholds` is not a debug property if 99% of people will decide to keep using default values. The current list of `9` package managers is not exhaustive, there are more package managers in the world. This module has to be extensible with new commands, via official ways, not some hidden secret properties.

Speaking of commands I use, `aur vercmp-devel` is currently a `contrib` script, not part of official aurutils distribution, so not generic.